### ISSUE TITLE
World locations

### ISSUE THREAD

**Issue by tukkek** [tukkek/javelin#56]

**Comment by tukkek** [comment]
2.0
	arena (unique) - offer a selection of unique monsters to fight, nonlethal
	Arcane University: upgrade your spells' caster level (touch to ranged and so on)
		make sure all spells are as close as possible to the books
		
	non-guarded, sacrificeable: resources.  Represents 1 labor.
	    Should be based on terrain:
	        wood on forest - never in autumn
	        ore on mountain -
	        fish on sea - never in winter, double in summer
	        game on plains - double in summer
	        crops on hill - never in winter, double on sprng
	        desert/marsh - never anything
	    Squads should carry labor for this and deforestate
	        can either return it to a town or use it to fasten Work improvements
	        should add resource to Troves as well?
	    once per turn select a random free terrain tile
	        try to generate a resource (see above)
	        base chance 1/7 (once per week)

**Comment by tukkek** [comment]
Dragon spire (dragon breath upgrades). Single Dragon Guardian?
	make breath effects work (possibly reactivate magma)
	use new BattlePanel#overlay to return overlar for breaths?
	make breaths a Target instead of directional?
        limit maximum quantity of breaths? (2?)
        reactivate breaths for ai auto-upgrade
        
it should be possible to sacrifice for hordes of skeletons in the Pillar of skulls (the lifeforce allows the skeletons to get out of the pillar, equal EL, using most powerful skeletong type)

crystal mountain: a place where you can trade rubies for money. x rubies will get you  $
    "the cave's bishop offers to trade your x magical rubies for crystals worth $treasure(el=x)"
    "the cave's bishop doesn't receive you, for you have no rubies that would interest him"

faerie ring: throw in rubies for a random magic item in return (item has a 50% chance of appearing per day after rubies are paid)

oasis (non unique but max of 1): 50% heals your wounds, 25% chance is a mirage, 25% chance poisons you

monolith: pay 1 ruby to reveal a random 4x4 section of the map
    a spell/shrine ritual probably already does it

Maybe haxor should be spread into several location similar to some proposed here? Maybe win the game automatically upon retrieving the 5 relics? (checked upon pickup)

tree of knowledge: trades rubies for experience? 1 ruby for 10XP?

tree of life: could take some functions from haxor liek rebirth

wishing well: could take some of haxor's functions?

tree of knowledge
    show entire world map for 20 ruubies

fountain of youth
    unique location
    grab things from haxor like rejuvenation and forget ability
    
Stronghold
    massive battle where the player is aided by a big number of guards
    unlike battlefield this can only be done in-person
    would be interesting if this was a "capture first" and "defend later" scenario - possibly with it being recaptured if not defended in a certain amount of time, etc
    also interesting would be if the levels escalated, let's say first a level 5 capture, then a level 6 defend, etc

**Comment by tukkek** [comment]
Great library: (hostile, level 10-15, unique) an abandoned Academy where knowledgeable players can research the spells within. Carries 10 random spells, each spell taking 1 week per level to decipher.  Show a prompt with all available spells and how much time it takes to decipher them. Needs at least DC15+level in spellcraft to start research. Once started absorbs the sage and includes the upgrade once done through turn(). Can have one or more spells being studied at the same time, by different researchers.

Adds a submenu r (research) to the academy screen

**Comment by tukkek** [comment]
Ecumenical temple

    Offers altars - each of them you sacrifice something in return for a buff or permanent effect
    Not sure if this isn’t already covered by normal Shrines
    Which gods?

A good implementation would be for an altar to take 1 ruby and just upgrade you on a selection, like kits. It uses a unit's entire XP pool. This is not as interesting as individual mechanics but it's pretty simple to implement and can even be used as individual altars for towns (that are normal academis). It's also different because the upgrade in the Temple's case is instantaneous.

Should have a maximum of 9 altars per game, even if there are more altars than that available. Could start with less though (5).

Once this is done, Religious towns should be able to build one (and only one) Altar in town as that of their favored deity.

**Comment by tukkek** [comment]
Would be cool if it was possible to buff attributes (+2 to score for 1 ruby) in 6 unique locations around the world - not dedicated but thematic ones.

**Comment by tukkek** [comment]
Unique dungeons (could all share a same icon, maybe a golden stair; probably better if they are fixed levels 5/10/15/20, randomly)

    Hall of mirrors (there are 10 interlinked mirrors that teleport the squad to another mirror in a random sequence; 50% of mirrors being “active” and enterable, renew with each mirrorsteping with at least 1 active at all times)
    Hall of cards (10 cards along the dungeon randomly remove adjacent walls, place discovered traps all around the squad or create an adjacent treasure chest)
    Rubik’s hall (a non-border wall moves into a random direction with each step)
    Jigsaw’s hall (no random encounters but tons of traps)

**Comment by tukkek** [comment]
I don’t think there’s a location yet where the squad joins another group in a bigger fight. This would be a cool pretext for a “Fort” - currently occupied by X race, do you want to help Y race (or possibly a few options) capture it? Once it’s captured (or even before) you can recruit as if in a Lair.

**Comment by tukkek** [comment]
Seer of Visions (icon: crystal ball): takes rubies in exchange for showing vision
1 to show a random location
2 to show nearest location
5 to show a district
9 to reveal whole map

**Comment by tukkek** [comment]
Could have a location that lets players alter personality traits for 1 ruby. Allow for a certain 3 steps, each offering 3 "replace trait with trait" or "replace flaw with flaw". Also have a "I'm done now" option and "Show me more options" (which expires on the last step). The Tree of Knowledge sounds like an OK place for this, but sounds like it would fit better at a crazy lab or something weird like that.

**Comment by tukkek** [comment]
It should be able to bring back "capture orbs". Make a location that lets you buy 5/10/15 orbs for gold - that you can then "refill" with XP. Or instead just have them sell a single orb but the maximum refill is based on the building upgrade level.

Capturing a monster subtracts the appropriate XP from the orb and has a % chance to succeed that is 0% at full HP and 100% at zero HP.
